modules/apps/notifications/notifications-web/src/main/java/com/liferay/notifications/web/internal/servlet/taglib/util/UserNotificationEventActionDropdownItem.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.notifications.web.internal.servlet.taglib.util;
+
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItem;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemListBuilder;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.model.UserNotificationEvent;
+import com.liferay.portal.kernel.notifications.UserNotificationFeedEntry;
+import com.liferay.portal.kernel.portlet.url.builder.PortletURLBuilder;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.subscription.model.Subscription;
+import com.liferay.subscription.service.SubscriptionLocalServiceUtil;
+
+import java.util.List;
+
+import javax.portlet.RenderResponse;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Eudaldo Alonso
+ */
+public class UserNotificationEventActionDropdownItem {
+
+	public UserNotificationEventActionDropdownItem(
+		HttpServletRequest httpServletRequest, RenderResponse renderResponse) {
+
+		_httpServletRequest = httpServletRequest;
+		_renderResponse = renderResponse;
+	}
+
+	public List<DropdownItem> getActionDropdownItems(
+			UserNotificationEvent userNotificationEvent,
+			UserNotificationFeedEntry userNotificationFeedEntry)
+		throws Exception {
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)_httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		JSONObject jsonObject = JSONFactoryUtil.createJSONObject(
+			userNotificationEvent.getPayload());
+
+		long subscriptionId = jsonObject.getLong("subscriptionId");
+
+		if (subscriptionId > 0) {
+			Subscription subscription =
+				SubscriptionLocalServiceUtil.fetchSubscription(subscriptionId);
+
+			if (subscription == null) {
+				subscriptionId = 0;
+			}
+		}
+
+		long finalSubscriptionId = subscriptionId;
+
+		return DropdownItemListBuilder.add(
+			() ->
+				!userNotificationEvent.isActionRequired() &&
+				!userNotificationEvent.isArchived(),
+			dropdownItem -> {
```

Comment:
Did you test if  all these cases work with just `dropdownItem.setHref(...URL)`? Unless I am missing something, we should not need the `putData("action", ...)` in these cases nor the `propsTransformer`.

Reply:
Tested and works with `setHref(...)`, I will change it.